code/modules/recycling/loafer.dm

```diff
@@ -0,0 +1,219 @@
+//Prison Loafer//
+
+/obj/structure/disposalpipe/loafer
+	name = "disciplinary loaf processor"
+	desc = "A pipe segment designed to convert detritus into a nutritionally-complete meal for inmates."
+	icon = 'icons/goonstation/objects/machinery.dmi'
+	icon_state = "pipe-loaf0"
+
+/obj/structure/disposalpipe/loafer/horizontal
+	dir = EAST
+
+/obj/structure/disposalpipe/loafer/vertical
+	dir = NORTH
+
+/obj/structure/disposalpipe/loafer/New()
+	..()
+	dpdir = dir | turn(dir, 180)
+	update()
+
+/obj/structure/disposalpipe/loafer/transfer(obj/structure/disposalholder/H)
+	if(H.contents.len)
+		playsound(loc, 'sound/goonstation/machines/mixer.ogg', 50, 1)
+		icon_state = "pipe-loaf1"
+
+		var/doSuperLoaf = 0
+		for(var/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/PL in H)
+			if(PL.super_dense)
+				doSuperLoaf = 1
+				break
+
+		if(doSuperLoaf)
+			for(var/atom/movable/O2 in H)
+				qdel(O2)
+
+			var/obj/item/weapon/reagent_containers/food/snacks/einstein_loaf/estein = new /obj/item/weapon/reagent_containers/food/snacks/einstein_loaf(src)
+			estein.forceMove(H)
+			loafer_pause()
+			return ..()
+
+		var/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/newLoaf = new /obj/item/weapon/reagent_containers/food/snacks/prison_loaf(src)
+		for(var/atom/movable/newIngredient in H)
+			if(newIngredient.reagents)
+				newIngredient.reagents.trans_to(newLoaf, 1000)
+
+			if(istype(newIngredient, /obj/item/weapon/reagent_containers/food/snacks/prison_loaf))
+				var/obj/item/weapon/reagent_containers/food/snacks/prison_loaf/otherLoaf = newIngredient
+				newLoaf.loaf_factor += otherLoaf.loaf_factor * 1.2
```

Comment:
Won't this let you just set this contraption in a cycle, growing at a rate of 1.2^x each time?